hcl/hcl_script.go

```diff
@@ -0,0 +1,148 @@
+	ErrScriptUnrecognizedAttr    errors.Kind = "terramate schema error: (script): unrecognized attribute"
+	ErrScriptUnrecognizedBlock   errors.Kind = "terramate schema error: (script): unrecognized block"
+	ErrScriptNoCmds              errors.Kind = "terramate schema error: (script): missing command or commands"
+	ErrScriptMissingOrInvalidJob errors.Kind = "terramate schema error: (script): missing or invalid job"
+	ErrScriptCmdConflict         errors.Kind = "terramate schema error: (script): conflicting attribute already set"
+)
+
+// Command represents an executable command
+type Command ast.Attribute
+
+// Commands represents a list of executable commands
+type Commands ast.Attribute
+
+// ScriptJob represent a Job within a Script
+type ScriptJob struct {
+	Command  *Command  // Command is a single executable command
+	Commands *Commands // Commands is a list of executable commands
+}
+
+// ScriptDescription is human readable description of a script
+type ScriptDescription ast.Attribute
+
+// Script represents a parsed script block
+type Script struct {
+	Labels      []string           // Labels of the script block used for grouping scripts
+	Description *ScriptDescription // Description is a human readable description of a script
+	Jobs        []*ScriptJob       // Job represents the command(s) part of this script
+}
+
+// NewScriptDescription returns a *ScriptDescription encapsulating an ast.Attribute
+func NewScriptDescription(attr ast.Attribute) *ScriptDescription {
+	desc := ScriptDescription(attr)
+	return &desc
+}
+
+// NewScriptCommand returns a *Command encapsulating an ast.Attribute
+func NewScriptCommand(attr ast.Attribute) *Command {
+	cmd := Command(attr)
+	return &cmd
+}
+
+// NewScriptCommands returns *Commands encapsulating an ast.Attribute
+func NewScriptCommands(attr ast.Attribute) *Commands {
+	cmds := Commands(attr)
+	return &cmds
+}
+
+func (p *TerramateParser) parseScriptBlock(block *ast.Block) (*Script, error) {
+	errs := errors.L()
+
+	parsedScript := &Script{
+		Labels: block.Labels,
+	}
+
+	for _, attr := range block.Attributes {
+		switch attr.Name {
+		case "description":
+			parsedScript.Description = NewScriptDescription(attr)
+		default:
+			errs.Append(errors.E(ErrScriptUnrecognizedAttr, attr.NameRange))
+		}
+	}
+
+	for _, nestedBlock := range block.Blocks {
+		switch nestedBlock.Type {
+		case "job":
+			parsedJobBlock, err := validateScriptJobBlock(nestedBlock)
+			if err != nil {
+				errs.Append(err)
+				continue
+			}
+			parsedScript.Jobs = append(parsedScript.Jobs, parsedJobBlock)
+		default:
+			errs.Append(errors.E(ErrScriptUnrecognizedBlock, nestedBlock.TypeRange, nestedBlock.Type))
+
+		}
+	}
+
+	if len(parsedScript.Labels) == 0 {
+		errs.Append(errors.E(ErrScriptNoLabels, block.TypeRange))
+	}
+
+	if len(parsedScript.Jobs) == 0 {
+		errs.Append(errors.E(ErrScriptMissingOrInvalidJob, block.OpenBraceRange))
```

Comment:
I think `block.Range` must be used here so it's clear that the `script` body is missing a `job` definition.

Reply:
Fixed in https://github.com/terramate-io/terramate/pull/1236/commits/0bd145bc88f507782997876f06b14dc3a983a24c